### PR TITLE
try/catch hotfix on indivdual getTransactionFor in router

### LINKED ISSUE #735: [router] Prevent ReceiverNotConfigured transactions from blocking other active transactions from being handled
ReceiverNotConfigured is blocking other active transactions from being handled on a chain, because router's subgraph peripheral was throwing when the receiving chain does not have a configured subgraph.

See https://github.com/connext/nxtp/issues/732. This is the underlying issue that exposed this problem as well.

### PR DESCRIPTION
**Pull request by base0010**
## The Solution

Fixes #735 

## Checklist

- [ ] Test manually on `test-ui` using remote chains.
- [ ] Run load tests.
- [ ] Update documentation if needed.
- [ ] Update CHANGELOG.md.

### CODE REVIEW

File: packages/router/src/adapters/subgraph/subgraph.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
   SubgraphSyncRecord,
   VariantTransactionData,
 } from "@connext/nxtp-utils";
-import { BigNumber, constants } from "ethers";
+import { BigNumber, constants, logger } from "ethers";
```
Comment (LayneHaber) [inline packages/router/src/adapters/subgraph/subgraph.ts:12]: Wrong logger, it should be the one from the utils package

File: packages/router/src/adapters/subgraph/subgraph.ts [hunk 2]
```diff
@@ -488,46 +488,48 @@ export const getTransactionForChain = async (
   if (!sdk) {
     throw new ContractReaderNotAvailableForChain(chainId, { method, methodId });
   }
```
Comment (LayneHaber) [inline packages/router/src/adapters/subgraph/subgraph.ts:490]: If the problem is the sdk is not configured (see #735 ) will the added try-catch fix it?

File: packages/router/src/adapters/subgraph/subgraph.ts [hunk 3]
```diff
@@ -488,46 +488,48 @@ export const getTransactionForChain = async (
...
+          },
+          encryptedCallData: transaction.encryptedCallData,
+          encodedBid: transaction.encodedBid,
+          bidSignature: transaction.bidSignature,
+          signature: transaction.signature,
+          relayerFee: transaction.relayerFee,
+        }
+      : undefined;
+  } catch (e) {
+    logger.debug(`Problem getting transactions on chain ${chainId}\n Error ${e} `);
```
Comment (LayneHaber) [inline packages/router/src/adapters/subgraph/subgraph.ts:530]: should log with the `@connext/nxtp-utils` logger and but things like the `error` in the context like:
```ts
{
  error: jsonifyError(e),
}
```

**Review by LayneHaber — CHANGES_REQUESTED** [review]
Also missing corresponding unit tests

**Comment by sanchaymittal** [comment]
closing this pr, because of inactivity. Please reopen it if necessary.

**Pull request closed without merging**